XenAdmin/SettingsPanels/GpuEditPage.cs

```diff
@@ -322,38 +250,119 @@ public void ShowHideWarnings()
 
                 imgHA.Visible = labelHA.Visible = true;
 
-                labelGpuType.Enabled = comboBoxGpus.Enabled = false;
+                addButton.Enabled = deleteButton.Enabled = false;
                 return;
             }
 
-            labelGpuType.Enabled = comboBoxGpus.Enabled = true;
-            GpuTuple tuple = comboBoxGpus.SelectedItem as GpuTuple;
+            addButton.Enabled = !VGpus.Any(v => (Connection.Resolve(v.type)).compatible_types_in_vm.Count() == 0);

```

Comment:
I prefer to enhance this line like: 
addButton.Enabled = VGpus.All(v => { var x = (Connection.Resolve(v.type)); return x != null && x.compatible_types_in_vm.Count() > 0; });
"Supporting multiple vGPUs or not" is usually used to describe a type rather than a vGPU.